### PR TITLE
[RHPAM-3591] - Persist KIE Server maven and kie repositories

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>

### CODE REVIEW

**Comment by spolti** [comment]
@swati-kale Hi, I have updated the form as we discuss today earlier, can you please review?

**Comment by swati-kale** [comment]
> @swati-kale Hi, I have updated the form as we discuss today earlier, can you please review it?

Looks good now.

**Comment by spolti** [comment]
@desmax74 f.y.i. please take a look at the UI changes.

**Comment by vaibhavjainwiz** [comment]
@spolti 
I far I understood this PR, we has mapped a persistence volume with kie-server pod from where kie-cloud-operator downloads maven aritifacts.
I have seen 2 problem with approach :

- Concurrent access to shared memory
- Growing size of maven repo

We should address these 2 issues before merging this PR.

embedded_maven_repo

I have one more question in mind. Why kie-cloud-operator maintaing this maven repo? Doesn't it user responsiblity to maintain Nexux maven repository near to k8s cluster and provide maven mirror url to kie-server?

File: config/7.11.0/common.yaml [hunk 1]
```diff
@@ -1009,6 +1025,39 @@ servers:
                 #[[end]]
                 #[[end]]
       ## KIE server deployment config END
+    ## KIE server pvc BEGIN
+    #[[if .PersistRepos]]
+    persistentVolumeClaims:
+      - metadata:
+          name: "[[$.ApplicationName]]-m2-repository-claim"
```
Comment (vaibhavjainwiz) [inline config/7.11.0/common.yaml:1032]: Persistent volume to store maven artifact is maintained for every `ApplicationName`. Does it possible to have multiple `ApplicationName`? If so does both applications maintain their own maven repo?

File: pkg/controller/kieapp/defaults/defaults.go [hunk 2]
```diff
@@ -1676,6 +1691,14 @@ func isImmutable(cr *api.KieApp) bool {
 	return false
 }
 
+func isTrial(cr *api.KieApp) bool {
```
Comment (vaibhavjainwiz) [inline pkg/controller/kieapp/defaults/defaults.go:1694]: IsTrial function uses status of `KieApp` CR. CR status will not be available at this point.

File: vendor/github.com/RHsyseng/console-cr-form/pkg/web/form.go [hunk 3]
```diff
@@ -6,6 +6,7 @@ type Form struct {
 
 type Page struct {
 	Label    string   `json:"label,omitempty"`
+	Visible  bool     `json:"visible,omitempty"`
```
Comment (vaibhavjainwiz) [inline vendor/github.com/RHsyseng/console-cr-form/pkg/web/form.go:9]: Does this change related to this story?

File: vendor/github.com/RHsyseng/console-cr-form/pkg/web/form.go [hunk 4]
```diff
@@ -22,6 +23,7 @@ type Field struct {
 	Max              *int     `json:"max,omitempty"`
 	OriginalJSONPath string   `json:"originalJsonPath,omitempty"`
 	Visible          bool     `json:"visible,omitempty"`
+	Checked          bool     `json:"checked,omitempty"`
```
Comment (vaibhavjainwiz) [inline vendor/github.com/RHsyseng/console-cr-form/pkg/web/form.go:26]: same as above

File: config/7.11.0/common.yaml (hunk 1 above)
Comment (spolti) [reply config/7.11.0/common.yaml:1032]: each deployment config should point to its own PV.

File: pkg/controller/kieapp/defaults/defaults.go (hunk 2 above)
Comment (spolti) [reply pkg/controller/kieapp/defaults/defaults.go:1694]: not really, it is part of the reconciliation loop. 
the CR is available, as well the CR.Applied confs.

**Comment by spolti** [comment]
replaced by https://github.com/kiegroup/kie-cloud-operator/pull/671

**Pull request closed without merging**